examples/all-devices-app/linux/device_impl/OccupancySensorDeviceImpl.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "OccupancySensorDeviceImpl.h"
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <platform/CHIPDeviceLayer.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+
+namespace {
+constexpr uint16_t kOccupancyStateChangeInterval = 30; // seconds
+} // namespace
+
+OccupancySensorDeviceImpl::OccupancySensorDeviceImpl() :
+    OccupancySensorDevice(
+        // Initialize with kInvalidEndpointId. The actual endpoint ID will be set
+        // when Register() is called by the application with a valid endpoint ID.
+        chip::app::Clusters::OccupancySensingCluster::Config(kInvalidEndpointId)
+            .WithFeatures(OccupancySensing::Feature::kPassiveInfrared)
+            .WithHoldTime(10,
+                          {
+                              .holdTimeMin     = 1,
+                              .holdTimeMax     = 300,
+                              .holdTimeDefault = 10,
+                          },
+                          mTimerDelegate)
+            .WithDelegate(this),
+        mTimerDelegate)
+{
+    // Kick off the timer loop to flip occupancy every few seconds
+    LogErrorOnFailure(mTimerDelegate.StartTimer(this, System::Clock::Seconds16(kOccupancyStateChangeInterval)));
```

Comment:
Do we expect this device to be in an operable state if the timer fails? Should we have a verify or die here? Or maybe have the timer start in an init or startup method? 

I know this is for the demo but one might expect if we don't have timers working at this point something is wrong.